### PR TITLE
Adds Bleach Reagent & Reaction

### PR DESCRIPTION
**Pull request by VaelophisNyx**
5eq87u

# Document the changes in your pull request

Adds Bleach, made in a reaction of 1u table salt + 2u water + 1u sterilizine cooled to 188K, to produce 4u of Bleach
a ghetto reaction exists that costs 1u of table salt, 2u of plasma, 1u liquid gibs, and 3u of ice, but does not require cooling. also produces 4u (less efficient)

Bleach is a strong poison but also a strong sterilizer
its effects are as follows:
- Mild toxin and stamina damage on a 1/3 chance per tick, which also makes you vomit (no stun) and lose your breath
- a 1/3 chance to deal light damage to your Stomach, Lungs, or Brain, 1/3 chance each if this is chosen. you also lose your breath a bit
- a 1/3 chance to roll for a 3% chance to cure ALL diseases in you, but not give you antibodies.

it has an overdose threshold of 30u
overdosing does the following:
- you lose your breath constantly
- your stomach takes mild damage every tick
- a 20% chance roll for heavier breath loss, very heavy toxin damage, a little burn damage, and blood vomit which DOES stun you.

Overdose of Bleach is **Very** hard to survive.

# Wiki Documentation

Must be added to the Guide to Chemistry

# Changelog

:cl:  
rscadd: Added Bleach and a way to create it
/:cl:

### CODE REVIEW

**Comment by Moltijoe** [comment]
now make it react with ammonia

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -940,6 +940,34 @@
...
+	if(method == TOUCH || method == VAPOR)
+		M.wash(clean_types)
+	if(method in list(TOUCH, VAPOR))
+		for(var/s in C.surgeries)
+			var/datum/surgery/S = s
+			S.success_multiplier = max(0.1, S.success_multiplier)
+			// +10% success propability on each step, useful while operating in less-than-perfect conditions
+	..()
+
+/datum/reagent/space_cleaner/bleach/on_mob_life(mob/living/carbon/M) //stronger cyanide, basically
```
Comment (redmoogle) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
/datum/reagent/space_cleaner/bleach/on_mob_life(mob/living/carbon/C) //stronger cyanide, basically
```

**Comment by neeshacark** [comment]
make it so spacecleaner only works on dirt and stuff and you need bleach to get bloodstains out

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -940,6 +940,43 @@
...
+			M.adjustOrganLoss(ORGAN_SLOT_BRAIN, 1 * REM)
+	if(prob(10))
+		to_chat(M, "You feel horrendously weak!")
+		M.adjustStaminaLoss(4*REM, 0)
+		M.adjustToxLoss(3.25*REM, 0)
+		M.vomit(0, FALSE, FALSE, 3)
+	if(prob(1))
+		for(var/thing in M.diseases)
+			var/datum/disease/D = thing
+			D.cure(FALSE)
```
Comment (tattax) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
		for(var/datum/disease/D as anything in M.diseases)
			D.cure(FALSE)
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 3]
```diff
@@ -940,6 +940,43 @@
...
+/datum/reagent/space_cleaner/bleach/reaction_mob(mob/living/carbon/C, method=TOUCH, reac_volume)
+	. = ..()
+	if(method in list(TOUCH, VAPOR))
+		for(var/s in C.surgeries)
+			var/datum/surgery/S = s
+			S.success_multiplier = max(0.1, S.success_multiplier)
+			// +10% success propability on each step, useful while operating in less-than-perfect conditions
+	..()
+
+/datum/reagent/space_cleaner/bleach/on_mob_life(mob/living/carbon/M) //stronger cyanide, basically
```
Comment (tattax) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: it's stronger cyanide but it's easier to make. Doesn't make much sense

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 4]
```diff
@@ -940,6 +940,43 @@
...
+	description = "A harsh chemical sterilizer. Do Not Ingest."
+	color = "#fbfdd6"
+	taste_description = "burning"
+	toxpwr = 0 //all the damage comes from the life process
+
+/datum/reagent/space_cleaner/bleach/reaction_mob(mob/living/carbon/C, method=TOUCH, reac_volume)
+	. = ..()
+	if(method in list(TOUCH, VAPOR))
+		for(var/s in C.surgeries)
+			var/datum/surgery/S = s
```
Comment (tattax) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
		for(var/datum/surgery/S as anything in C.surgeries)
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 3 above)
Comment (VaelophisNyx) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: unfortunately the chems I *would* need for it to be harder and more realistic, are actually used by similar reactions or not in the code.

Comment (VaelophisNyx) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: also chilling outside of chem is difficult to pull off

**Comment by boodaliboo** [comment]
> make it so spacecleaner only works on dirt and stuff and you need bleach to get bloodstains out

no thats stupid af and would make slaughter demons significantly more powerful

**Comment by CMOisLing** [comment]
You should make it not do brain and lung damage. That doesnt really fit with the theme of how toxins are done. For example Cyanide doesnt cause brain or heart damage despite its irl counterpart doing exactly that.

**Comment by VaelophisNyx** [comment]
> You should make it not do brain and lung damage. That doesnt really fit with the theme of how toxins are done. For example Cyanide doesnt cause brain or heart damage despite its irl counterpart doing exactly that.

so maybe we should buff cyanide

**Comment by Bibby0110** [comment]
Seems unnecessary tbh

**Comment by VaelophisNyx** [comment]
>server begs for new content
>new content is made
>"seems unnecessary"

☠

**Comment by JamieD1** [comment]
Pointless bloat

**Pull request closed without merging**